lmp/preload-app-images

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3
+#
+# Copyright (c) 2020 Foundries.io
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import argparse
+import logging
+import json
+import os
+import subprocess
+
+
+logging.basicConfig(level='INFO')
+fh = logging.FileHandler('/archive/preload-app-images.log')
+fh.setFormatter(logging.getLogger().handlers[0].formatter)
+logging.getLogger().addHandler(fh)
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('installed_versions')
+    parser.add_argument('dst_dir')
+    parser.add_argument('arch')
+    return parser.parse_args()
+
+
+def preload_app_images(target_json_file_filepath, dst_dir, arch):
+    with open(target_json_file_filepath) as target_json_file:
+        target_json = json.load(target_json_file)
+
+    target_name = list(target_json.keys())[0]
+    logging.info('Preloading app images for ' + target_name)
+    sha = target_json[target_name]['custom']['containers-sha']
+    image_data_dir = os.path.join('/var/cache/bitbake/app-images/', sha)
+
+    arch_map = {'aarch64': 'arm64', 'x86_64': 'amd64', 'arm': 'arm'}
+    image_data_tar = os.path.join(image_data_dir, '{}-{}.tar'.format(sha[:7], arch_map[arch]))
```

Comment:
just noticed this - `sha[:7]` isn't actually always the case:
  https://github.com/foundriesio/ci-scripts/commit/8edbdad88b882d91bd4b1316c7197c071ef8d0fd

Reply:
Are you implying to use a full hash/sha value? BTW, we use it in other places, e.g. https://github.com/foundriesio/ci-scripts/blob/321d877d80f560b24ef99f221685e1b2a341f231/factory-containers/publish-compose-apps#L112 (tag is 7 symbol hash).

Reply:
not the full hash but tag needs to be computed correctly. In the example you shared its actually getting that from the TAG environment variable set in docker-app-publish.sh which uses `git log -1 --format=%h`. Basically what I'm saying is 99% of the time the short-hash (TAG) is 7 characters. However, we occasionally have exceptions, so you have to use `git log -1 --format=%h` to ensure you get the correct tag.

Reply:
understood. In this case, I should think of something else as the given script is called during lmp build while 'sha' is a sha of corresponding containers repo commit so running `git log -1 --format=%h` will yield an abbreviated hash of the current lmp commit.

Reply:
updated